### PR TITLE
Remove jQuery from the "find file" page

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched to plain JavaScript
- Tested the file searching functionality and it works as before

# Demo using JavaScript without jQuery
action

### CODE REVIEW

**Review by delvh — APPROVED** [review]

File: web_src/js/features/repo-findfile.js [hunk 1]
```diff
@@ -74,46 +72,41 @@ export function filterRepoFilesWeighted(files, filter) {
...
     // if the target file path is "abc/xyz", to search "bx", then the matchResult is ['a', 'b', 'c/', 'x', 'yz']
     // the matchResult[odd] is matched and highlighted to red.
-    for (let j = 0; j < r.matchResult.length; j++) {
-      if (!r.matchResult[j]) continue;
-      const $span = $('<span>').text(r.matchResult[j]);
-      if (j % 2 === 1) $span.addClass('ui text red');
-      $a.append($span);
-    }
-    $repoFindFileTableBody.append($row);
+      `<span class="${index % 2 === 1 ? 'ui text red' : ''}">${part}</span>`
```
Comment (silverwind) [inline web_src/js/features/repo-findfile.js]: Is any user content in `part` correctly escaped?

File: web_src/js/features/repo-findfile.js [hunk 2]
```diff
@@ -74,46 +72,41 @@ export function filterRepoFilesWeighted(files, filter) {
...
-    const $row = $(tmplRow);
-    const $a = $row.find('a');
-    $a.attr('href', `${treeLink}/${pathEscapeSegments(r.matchResult.join(''))}`);
-    const $octiconFile = $(svg('octicon-file')).addClass('gt-mr-3');
-    $a.append($octiconFile);
+    const row = document.createElement('tr');
+    const cell = document.createElement('td');
+    const a = document.createElement('a');
+    a.setAttribute('href', `${treeLink}/${pathEscapeSegments(r.matchResult.join(''))}`);
+    a.innerHTML = svg('octicon-file') + r.matchResult.map((part, index) =>
```
Comment (silverwind) [inline web_src/js/features/repo-findfile.js]: Do we need the `gt-mr-3` on the svg? If yes, use `svg('octicon-file', 16, 'gt-mr-3)`.

File: web_src/js/features/repo-findfile.js (hunk 1 above)
Comment (wxiaoguang) [reply web_src/js/features/repo-findfile.js]: No for the changed code. So possible XSS ....

Comment (yardenshoham) [reply web_src/js/features/repo-findfile.js]: Is it OK now with textContent?

Comment (silverwind) [reply web_src/js/features/repo-findfile.js]: Yes, `.textContent` is always safe.

**Review by silverwind — APPROVED** [review]

File: web_src/js/features/repo-home.js [hunk 3]
```diff
@@ -52,12 +54,14 @@ export function initRepoTopicBar() {
...
       }
-    }).fail((xhr) => {
+    } catch (error) {
+      const xhr = error.response; // Assuming `error.response` contains the fetch response object
       if (xhr.status === 422) {
-        if (xhr.responseJSON.invalidTopics.length > 0) {
-          topicPrompts.formatPrompt = xhr.responseJSON.message;
+        const responseData = await xhr.json(); // Parse JSON response
+        if (responseData.invalidTopics.length > 0) {
+          topicPrompts.formatPrompt = responseData.message;
```
Comment (wxiaoguang) [inline web_src/js/features/repo-home.js]: 1. Is this change related to "Remove jQuery from the "find file" page"?
2. Has this change been tested to see the proper error handling?

Comment (yardenshoham) [reply web_src/js/features/repo-home.js]: Sorry working on too many things at once, pushed this file by accident. It's not ready for review

**Pull request merged**